[Order Creation] Handle keyboard predictive text entry for country and state selectors

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #6644
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
When choosing a country or state when filling the customer address in an order, using the predictive text on the iPhone keyboard adds an empty space at the end of the search term.  This makes the filter unable to show the result for that word, even though the country/state exists.
With this PR we remove the empty space of the search term before actually filtering the list with it. This way the country/state is properly found.

### Changes
- ~~Add String helper function `removeLastCharacterIfWhitespace`~~ Not needed thanks to @Ecarrion comment. We can use the already existing one `trim()`
- Apply `trim()` to the search term to remove whitespace or newlines from both ends in `CountrySelectorCommand` and `StateSelectorCommand`
- Add unit test in `CountrySelectorViewModelTests`

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

(Prerequisite: Have an order with Customer Shipping Details)
1. Open App
2. Go to orders
3. Tap on Shipping Details
4. Tap on Country
5. Tap on Filter Countries Text Field
6. Add a country name, and then, use the keyboard predictive term entry to add it
7. List of countries is filtered showing that unique term. Tapping on it checks it.

### Screenshots
<!-- Include before and after images or gifs when appropriate. -->
#### Before
https://user-images.githubusercontent.com/266376/163336776-09c97235-5ae7-4396-aab0-d9ce35e768df.MP4
#### After
https://user-images.githubusercontent.com/1864060/163825926-355bf110-d731-45f8-8347-64c879b87b5d.mp4

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

toupper committed:
Add function to remove the last character of a string if it is a whitespace. Useful for search terms.

toupper committed:
Remove last whitespace if any from the search term, as it wouldn't be found in the suggestions list otherwise.

toupper committed:
Unit tests for removeLastCharacterIfWhitespace

toupper committed:
Add comments to clarify the reason behind using this helper function

Ecarrion reviewed (commented):
Hey @toupper, first of all, thanks for helping us with this issue! 🙇 

Second, I think we can use the `trim()` function available in `String+Helpers` which removes whitespace & new lines, this could help when people paste content.

Additionally, do you think you could add unit tests to `CountrySelectorViewModelTests`?

toupper committed:
Use existing trim function for search terms instead of using a new function removeLastCharacterIfWhitespace

toupper committed:
Remove not needed function

toupper committed:
Add unit tests for country selector for search terms with newlines or whitespaces at the end

toupper committed:
Add comment to clarify the reason behind trimming the search term

toupper commented:
@Ecarrion Thanks for your input! 
Indeed, we can use `trim` instead of creating a new function. I removed the not necessary code and apply `trim` as suggested. Furthermore, I added unit tests for these cases in `CountrySelectorViewModelTests` 🚀

Ecarrion reviewed (approved):
Amazing, Thanks! 🚀
